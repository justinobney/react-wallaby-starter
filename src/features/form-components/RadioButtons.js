import React, {Component} from 'react';
import {Button} from 'semantic-ui-react';
import P from 'prop-types';

export function Func() {}

class RadioButtons extends Component {
  static propTypes = {
    name: P.string.isRequired,
    onChange: P.func.isRequired,
  };

  onClick = (e, {value}) => {
    const {name, onChange} = this.props;
    onChange(e, {name, value});
  };

  render() {
    const {value, options} = this.props;
    return (
      <Button.Group>
        {options.map(x => (
          <Button
            type="button"
            key={x.label}
            active={x.value === value}
            onClick={e => this.onClick(e, x)}
          >
            {x.label}
          </Button>
        ))}
      </Button.Group>
    );
  }
}

export default RadioButtons;
