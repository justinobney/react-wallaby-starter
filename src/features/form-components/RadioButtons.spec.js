import React from 'react';
import {render, Simulate} from 'react-testing-library';
import RadioButtons from './RadioButtons';

describe('Component: RadioButtons', () => {
  it('default render with selected value', () => {
    const options = [
      {value: 5, label: '5'},
      {value: 10, label: '10'},
      {value: 15, label: '15'},
    ];

    const {container} = render(
      <RadioButtons
        name="test"
        options={options}
        value={5}
        onChange={Function()} // eslint-disable-line
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('default render without selected value', () => {
    const options = [
      {value: 5, label: '5'},
      {value: 10, label: '10'},
      {value: 15, label: '15'},
    ];

    const {container} = render(
      <RadioButtons options={options} name="test" onChange={Function()} /> // eslint-disable-line
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('clicking an option calls the callback', () => {
    const options = [
      {value: 5, label: '5'},
      {value: 10, label: '10'},
      {value: 15, label: '15'},
    ];

    const onChange = jest.fn();

    const {getByText} = render(
      <RadioButtons
        name="test"
        options={options}
        value={10}
        onChange={onChange}
      />
    );

    Simulate.click(getByText('5'));

    expect(onChange.mock.calls[0][1]).toEqual({name: 'test', value: 5});
  });
});
