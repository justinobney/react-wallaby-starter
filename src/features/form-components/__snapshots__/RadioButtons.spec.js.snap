// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RadioButtons default render with selected value 1`] = `
<div
  className="ui buttons"
>
  <button
    className="ui active button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
    type="button"
  >
    5
  </button>
  <button
    className="ui button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
    type="button"
  >
    10
  </button>
  <button
    className="ui button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
    type="button"
  >
    15
  </button>
</div>
`;

exports[`Component: RadioButtons default render without selected value 1`] = `
<div
  className="ui buttons"
>
  <button
    className="ui button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
    type="button"
  >
    5
  </button>
  <button
    className="ui button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
    type="button"
  >
    10
  </button>
  <button
    className="ui button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
    type="button"
  >
    15
  </button>
</div>
`;
